python/src/aiconfig/util/config_utils.py

```diff
@@ -10,10 +11,28 @@
     from ..schema import AIConfig
 
 
-def get_api_key_from_environment(api_key_name: str):
-    if api_key_name not in os.environ:
-        raise Exception("Missing API key '{}' in environment".format(api_key_name))
+def maybe_get_api_key_from_environment(
+    api_key_name: str,
+    required: bool = True) -> Union[str, None]:
```

Comment:
I don't think the "required" logic belongs here. This function should be responsible for 1 thing: get it if it's there, otherwise return None. The required or not branch should happen above this and just not call this function.

Also please type this as str | None. Union[str, None] is definitely not the standard. Optional[str] is the old but reasonable way.